install.sh

```diff
@@ -17,58 +14,161 @@ CYAN='\033[0;36m'
+		return 1
+	}
+	check_dep "stat" && {
+		if [ "$(stat -c "%G" $install_path)" = "root" ]; then
+			notset "$root" && root=1
+		fi
+	}
+}
+
+## fix when env var root=0 is set
+[ "$root" = "0" ] && root=
+
+## root text and else
+if notset "$root"; then
+	require_text="root not required"
+	sudo=
 else
-    url="https://raw.githubusercontent.com/anhsirk0/fetch-master-6000/master/fm6000.pl"
-    if [ "$(command -v curl)" ]; then
-        printf '%b\n' "${BLUE}Downloading the script${NC}"
-        curl $url -o fm6000
-    else
-        printf '%b\n' "${RED}curl is required${NC}" && exit 1
-    fi
+	require_text="root required"
 fi
 
-if [ "$(command -v doas)" ]; then
-    sudo=doas
+## set headless when it is not already set
+notset "$headless" && headless=
+
+## download fm6000 if it does not exist
+if [ -f "fm6000.pl" ]; then
+	out "${BLUE}Seems like you cloned the repository${NC}"
+	cp fm6000.pl fm6000
 else
-    sudo=sudo
+	url="https://raw.githubusercontent.com/anhsirk0/fetch-master-6000/master/fm6000.pl"
+	if check_dep "curl"; then
+		out "${BLUE}Downloading the script${NC}"
+		curl $url -o fm6000
+	else
+		out "${RED}curl is required${NC}" && return 1
+	fi
+fi
+
+if ! notset "$root"; then
+	## if user is root do not set sudo
+	if [ "$(id -u)" -eq "0" ]; then
+		sudo=
+	## check if doas is available
+	elif [ "$(command -v doas)" ]; then
+		sudo=doas
+	## if user is not root and doas is not found fallback to sudo
+	else
+		sudo=sudo
+	fi
 fi
 
 if [ -f "fm6000" ] && [ -s "fm6000" ]; then
-    chmod +x fm6000 && printf '%b\n' "${BLUE}Making the script executable : ${GREEN}done"
-    require_text="root required"
-    ans="$1"
-
-    if [ -x $install_path ]; then
-        if [ $root = 0 ]; then
-            require_text="root not required"
-            sudo=
-        fi
-        printf '%b' "${YELLOW}"
-        [ -z "$ans" ] && read -p "Move the script to $install_path [$require_text]? (y/N) " ans
-    else
-        install_path=/usr/local/bin
-        printf '%b' "${YELLOW}"
-        [ -z "$ans" ] && read -p "Move the script to $install_path [$require_text]? (y/N)  " ans
-    fi
-
-    if [ "${ans}" = "y" ] || [ "${ans}" = "-y" ]; then
-        printf '%b\n' "${BLUE}Moving fm6000 to $install_path${NC}"
-        printf '%b' "${CYAN}"
-        $sudo mv -v fm6000 $install_path || (
-            printf '%b\n' "${RED}error: $sudo failed${NC}"
-            exit
-            exit 1
-        ) # double exit. first one exits the function
-        printf '%b\n' "${GREEN}Fetch-master-6000 is successfully installed${NC}"
-        exit 0
-    else
-        printf '%b\n' "${YELLOW}Script not moved!${NC}"
-        ./fm6000
-        exit 0
-    fi
+	chmod +x fm6000 && out "${BLUE}Making the script executable : ${GREEN}done"
+
+	# shellcheck disable=SC2086
+	if check_dep "find" && [ "$(find /usr/bin /usr/local/bin $HOME/.local/bin -type f -iname 'fm6000')" ]; then
+		out "${RED}it seems fm6000 is already installed but continuing anyway${NC}"
+	fi
```

Comment:
Check if fm6000 is already available in /usr/bin or /usr/local/bin or $HOME/.local/bin